python_modules/dagster/dagster/_core/definitions/asset_daemon_context.py

```diff
@@ -107,14 +107,26 @@ def __init__(
         )
         self._data_time_resolver = CachingDataTimeResolver(self.instance_queryer)
         self._cursor = cursor
-        self._target_asset_keys = target_asset_keys or {
-            key
-            for key, policy in self.asset_graph.auto_materialize_policies_by_key.items()
-            if policy is not None
-        }
+        self._target_asset_keys = (
+            target_asset_keys
+            if target_asset_keys is not None
+            else {
+                key
+                for key, policy in self.asset_graph.auto_materialize_policies_by_key.items()
+                if policy is not None
+            }
+        )
         self._materialize_run_tags = materialize_run_tags
         self._observe_run_tags = observe_run_tags
-        self._auto_observe = auto_observe
+        self._auto_observe_asset_keys = (
+            auto_observe_asset_keys
+            if auto_observe_asset_keys is not None
```

Comment:
this is maybe not ideal in that it does a big behavior change between None and set() which is often a source of bugs - perhaps we should make it so that callsites just always need to pass in the keys that they want to include?

Reply:
I'm very pro-passing in the set of keys at all call sites for this reason.